apps/web/components/dialog/EditLocationDialog.tsx

```diff
@@ -0,0 +1,257 @@
+                id="address"
+                required
+                className="border-grays-300 block w-full rounded-sm text-sm shadow-sm "
+                defaultValue={
+                  defaultValues
+                    ? defaultValues.find(
+                        (location: { type: LocationType }) => location.type === LocationType.InPerson
+                      )?.address
+                    : formMethods
+                        .getValues("locations")
+                        .find((location: { type: LocationType }) => location.type === LocationType.InPerson)
+                        ?.address
+                }
+              />
+            </div>
+          </div>
+        );
+      case LocationType.Link:
+        return (
+          <div>
+            <label htmlFor="address" className="block text-sm font-medium text-gray-700">
+              {t("set_link_meeting")}
+            </label>
+            <div className="mt-1">
+              <input
+                type="text"
+                {...locationFormMethods.register("locationLink")}
+                id="address"
+                required
+                className="block w-full rounded-sm border-gray-300 shadow-sm sm:text-sm"
+                defaultValue={
+                  defaultValues
+                    ? defaultValues.find(
+                        (location: { type: LocationType }) => location.type === LocationType.Link
+                      )?.link
+                    : formMethods
+                        .getValues("locations")
+                        .find((location: { type: LocationType }) => location.type === LocationType.Link)?.link
+                }
+              />
+              {locationFormMethods.formState.errors.locationLink && (
+                <p className="mt-1 text-red-500">
+                  {locationFormMethods.formState.errors.locationLink.message}
+                </p>
+              )}
+            </div>
+          </div>
+        );
+      case LocationType.Phone:
+        return <p className="text-sm">{t("cal_invitee_phone_number_scheduling")}</p>;
+      /* TODO: Render this dynamically from App Store */
+      case LocationType.GoogleMeet:
+        return <p className="text-sm">{t("cal_provide_google_meet_location")}</p>;
+      case LocationType.Zoom:
+        return <p className="text-sm">{t("cal_provide_zoom_meeting_url")}</p>;
+      case LocationType.Daily:
+        return <p className="text-sm">{t("cal_provide_video_meeting_url")}</p>;
+      case LocationType.Jitsi:
+        return <p className="text-sm">{t("cal_provide_jitsi_meeting_url")}</p>;
+      case LocationType.Huddle01:
+        return <p className="text-sm">{t("cal_provide_huddle01_meeting_url")}</p>;
+      case LocationType.Tandem:
+        return <p className="text-sm">{t("cal_provide_tandem_meeting_url")}</p>;
+      case LocationType.Teams:
+        return <p className="text-sm">{t("cal_provide_teams_meeting_url")}</p>;
```

Comment:
There is a lib file that does almost the same as this switch, we can probably reuse it here but I'm not so sure since you have it here with some html elements.
`cal-com/cal.com/apps/web/lib/locationOptions.tsx`

```suggestion
        case default: 
             return <p className="text-sm">{LocationOptionsToString(location)}</p>;
             
```